### PR TITLE
Get vba macros

### PR DESCRIPTION
**Pull request by KhamZero**
Add functions to get and visualize VBA virus macros.

### CODE REVIEW

File: doc_analyzer.py [hunk 1]
```diff
@@ -145,25 +144,47 @@ def get_macros_infos(self):
...
             {'number': autoexec_and_base64,
-                'description': 'Автоматический вызов Base64 обфусцированных строк', 'danger': True},
+                'description': 'Автоматический вызов Base64 обфусцированных строк', 'danger': True, 'function': olevba.detect_base64_strings},
         ]
 
         vbaparser.close()
         return macros_infos
 
 
+    def get_vba_code(self):
```
Comment (govza) [inline doc_analyzer.py:166]: Please add description

File: doc_analyzer.py [hunk 2]
```diff
@@ -145,25 +144,47 @@ def get_macros_infos(self):
...
+        macros_infos = self.macros_infos
+        for i in range(len(macros_infos)):
+            if macros_infos[i]['number'] > 0:
+                macro_code = macros_infos[i]['function'](code)
+                for i in macro_code:
+                    for j in i:
+                        report_vba += j
+
+
+        return report_vba
```
Comment (govza) [inline doc_analyzer.py:189]: vbaparser.close() ?
also I don't like that it's created 2 times.

File: doc_analyzer.py [hunk 3]
```diff
@@ -145,25 +144,47 @@ def get_macros_infos(self):
...
+    def get_vba_code(self):
+        vbaparser = VBA_Parser(self.file_path)
+        vbaparser.detect_vba_macros()
+        code_list = list()
+
+        for (filename, stream_path, vba_filename, vba_code) in vbaparser.extract_macros():
+            code_list.append(vba_code)
+            code = vba_code
+
+        report_vba = code_list[0]
```
Comment (govza) [inline doc_analyzer.py:179]: No null check first? Basically it fails if file has no macros.
Also some message, i.e. no macros found?

**Review by govza — CHANGES_REQUESTED** [review]
Good work, some failures could be improved.

File: doc_analyzer.py (hunk 3 above)
Comment (KhamZero) [reply doc_analyzer.py:179]: ok, I agree

File: doc_analyzer.py (hunk 2 above)
Comment (KhamZero) [reply doc_analyzer.py:189]: 3 times actually...

Comment (KhamZero) [reply doc_analyzer.py:189]: But it's normal

Comment (govza) [reply doc_analyzer.py:189]: Not really DRY, but maybe we can improve this in next PR.

File: doc_analyzer.py [hunk 4]
```diff
@@ -145,27 +146,55 @@ def get_macros_infos(self):
...
+                    for j in i:
+                        report_vba += j
+
+        vbaparser.close()
+        return report_vba
+
+
 def check_one_file():
     """Check one file from documents directory."""
+
```
Comment (govza) [inline doc_analyzer.py]: whitespace not needed
No blank lines allowed after function docstring pydocstyle (D202)

File: doc_analyzer.py [hunk 5]
```diff
@@ -182,6 +211,7 @@ def check_one_file():
 
 def check_directory():
     """Check the directory files."""
+
```
Comment (govza) [inline doc_analyzer.py]: here also

**Review by govza — APPROVED** [review]

**Review by govza — APPROVED** [review]

**Pull request merged**